src/libsync/account.cpp

```diff
@@ -515,4 +515,26 @@ void Account::setServerVersion(const QString &version)
     emit serverVersionChanged(this, oldServerVersion, version);
 }
 
+QString Account::defaultSyncRoot() const
+{
+    Q_ASSERT(!_defaultSyncRoot.isEmpty());
+    return _defaultSyncRoot;
+}
+bool Account::hasDefaultSyncRoot() const
+{
+    return !_defaultSyncRoot.isEmpty();
```

Comment:
What's the motivation of having this extra function? Right now, one could just check whether the returned value is empty, too. We've done this elsewhere, too. In case there is a reason why you designed it like this, I'd appreciate some comments.

Reply:
See the comment about the asser.